Patch-Source: https://github.com/GothenburgBitFactory/taskwarrior/commit/2ee5fb287cf8bc28a4755b662c54ef4334be548e
Backport 3.4.1 regression fix for 3.4.0.
---
From 2ee5fb287cf8bc28a4755b662c54ef4334be548e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kalle=20Kiet=C3=A4v=C3=A4inen?=
 <1026741+kietavainen@users.noreply.github.com>
Date: Fri, 14 Mar 2025 02:01:16 +0200
Subject: [PATCH] Fix suppressing news nag after reading the news (#3817)

The news nag suppression regressed again in 5c67d22. That commit
intended to remove the sponsorship outro from the news, but also removed
the bookkeeping that marks the news as read. This commit reverts that
part back to its previous state.
---
 src/commands/CmdNews.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/commands/CmdNews.cpp b/src/commands/CmdNews.cpp
index cce5f079c..b6a6f75cb 100644
--- a/src/commands/CmdNews.cpp
+++ b/src/commands/CmdNews.cpp
@@ -588,6 +588,11 @@ int CmdNews::execute(std::string& output) {
   }
   wait_for_enter();
 
+  // Set a mark in the config to remember which version's release notes were displayed
+  if (news_version < current_version) {
+    CmdConfig::setConfigVariable("news.version", std::string(current_version), false);
+  }
+
   return 0;
 }
 
